src/fontra/views/editor/panel-reference-font.js

```diff
@@ -264,15 +287,115 @@ export default class ReferenceFontPanel extends Panel {
 
     this.controller.addKeyListener("charOverride", (event) => {
       this.editorController.canvasController.requestUpdate();
+      this.requestReferenceFontsPreview();
     });
 
     this.controller.addKeyListener("languageCode", (event) => {
       this.editorController.canvasController.setLangAttribute(this.model.languageCode);
+      this.requestReferenceFontsPreview();
     });
 
     this.editorController.canvasController.setLangAttribute(this.model.languageCode);
 
     referenceFontModel = this.model;
+
+    this.editorController.sceneSettingsController.addKeyListener(
+      ["selectedGlyph", "glyphLines"],
+      (event) => {
+        this.requestReferenceFontsPreview();
+      }
+    );
+  }
+
+  async requestReferenceFontsPreview() {
+    if (this.referenceFontsPreviewPromise) {
+      return this.referenceFontsPreviewPromise;
+    }
+    this.referenceFontsPreviewPromise = this.displayCurrentGlyphInReferenceFonts();
+    await this.referenceFontsPreviewPromise;
+    delete this.referenceFontsPreviewPromise;
+  }
+
+  async displayCurrentGlyphInReferenceFonts() {
+    const container = this.contentElement.querySelector(".reference-font-preview");
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+    const selectedGlyphInfo = getSelectedGlyphInfo(
+      this.editorController.sceneSettings.selectedGlyph,
+      this.editorController.sceneSettings.glyphLines
+    );
+
+    let textToDisplay;
+
+    if (this.model.charOverride) {
+      textToDisplay = this.model.charOverride.charAt(0);
+    } else {
+      if (selectedGlyphInfo) {
+        if (selectedGlyphInfo.glyphName.includes(".")) {
+          const baseGlyphName = selectedGlyphInfo.glyphName.split(".")[0];
+          const codePoint = (this.editorController.fontController.glyphMap[
+            baseGlyphName
+          ] || [])[0];
+          if (codePoint) {
+            textToDisplay = String.fromCodePoint(codePoint);
+          }
+        } else {
+          textToDisplay = selectedGlyphInfo.character;
+        }
+      }
+    }
+
+    if (!textToDisplay) {
+      return;
+    }
+
+    if (
+      !textToDisplay &&
+      selectedGlyphInfo &&
+      selectedGlyphInfo.glyphName.includes(".")
+    ) {
+      const baseGlyphName = positionedGlyph.glyphName.split(".")[0];
```

Comment:
This sections is almost identical to the above (at line 336). Why is this needed? It seems redundant.

Reply:
Yes, I forgot this section to delete when copying the behavior to the section above.